packages/xsnap/test/test-boot-lockdown.js

```diff
@@ -80,3 +80,23 @@ test('SES deep stacks work on xsnap', async t => {
   t.is(typeof msg, 'string');
   t.assert(msg.length >= 1);
 });
+
+test.only('TextDecoder under xsnap handles TypedArray and subarrays', async t => {
+  const bootScript = await asset('..', 'dist', 'bundle-ses-boot.umd.js');
+  const opts = options();
+  const vat = xsnap(opts);
+  await vat.evaluate(bootScript);
+  await vat.evaluate(`
+    const decoder = new TextDecoder();
+    const encoder = new TextEncoder();
+    const send = msg => issueCommand(encoder.encode(JSON.stringify(msg)).buffer);
+
+    const string = '0123456789';
+    const bytes = encoder.encode(string).subarray(1, 4);
```

Comment:
I'm a little fuzzy on how this test would fail without the fix in this PR. I suppose I should test it, but I'll take the risk that I may need to dig into this later instead.

Reply:
The test failed previously for a couple reasons. The first was that String.fromArrayBuffer threw an exception if you passed a Uint8Array (which is a TypedArray view of an ArrayBuffer). The Uint8Array has a `buffer` property you can use instead, but that would fail because String.fromArrayBuffer assumes the data to decode starts at byteOffset 0, but a TypeArray view can be a view into a window at a non-zero offset. So, this test ensures that the `subarray(1, 4)` causes truncates the buffer at an offset and that carries over as as equivalent to the change to the resulting string.

The latest XS already addresses both of these problems by making `String.fromArrayBuffer` accept `TypeArray` views into array buffers and also taking into account their `byteOffset` and `byteLength`. It does this without resorting to a memcopy, which is strictly better than this work-around.